Gem/Code/Source/RHI/VariableRateShadingExampleComponent.h

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Math/Color.h>
+#include <AzCore/std/containers/array.h>
+
+#include <AzFramework/Input/Events/InputChannelEventListener.h>
+#include <AzFramework/Windowing/WindowBus.h>
+#include <AzFramework/Windowing/NativeWindow.h>
+
+#include <Atom/RPI.Public/Shader/Shader.h>
+#include <Atom/RPI.Public/Shader/ShaderResourceGroup.h>
+
+#include <Atom/RHI/BufferPool.h>
+#include <Atom/RHI/DrawItem.h>
+#include <Atom/RHI/Device.h>
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/CopyItem.h>
+#include <Atom/RHI/FrameScheduler.h>
+#include <Atom/RHI/PipelineState.h>
+
+#include <RHI/BasicRHIComponent.h>
+#include <Utils/ImGuiSidebar.h>
+#include <Utils/ImGuiProgressList.h>
+#include <Atom/Utils/AssetCollectionAsyncLoader.h>
+
+namespace AtomSampleViewer
+{
+    //! This samples demonstrates the use of Variable Rate Shading on the RHI.
+    //! Shading rates can be specified in 3 different ways: PerDraw, PerPrimtive and PerRegion.
+    //! This samples only uses the PerDraw and PerRegion modes.
+    //! The samples render a full screen quad using different shading rates.
+    //! When the PerRegion mode is used, an image is generated using a compute shader with different
+    //! rates in a circular pattern from the center (or the pointer position).
+    //! When a PerDraw mode is used, the rate is applied equally to the whole quad. The rate can be changed
+    //! using the GUI of the sample.
+    //! Combinator operations are also exposed when both PerDraw and PerRegion are being used.
+    class VariableRateShadingExampleComponent final
+        : public BasicRHIComponent
+        , public AZ::TickBus::Handler
+        , public AzFramework::InputChannelEventListener
+    {
+    public:
+        AZ_COMPONENT(VariableRateShadingExampleComponent, "{B98E1C6A-8C23-4AA4-82E6-4B652F6151DD}", AZ::Component);
```

Comment:
I think the third parameter of `AZ_COMPONENT` if the base class, which in this case is `BasicRHIComponent`.